pageserver/src/tenant/timeline/walreceiver/connection_manager.rs

```diff
@@ -214,6 +228,52 @@ pub(super) async fn connection_manager_loop_step(
                     }
                 }
             } => debug!("Waking up for the next retry after waiting for {time_until_next_retry:?}"),
+
+            Some(()) = async {
```

Comment:
Just making sure, did you audit this code for cancellation safety?

Reply:
I tried, but honestly haven't found anything about [tonic](https://github.com/hyperium/tonic) cancellation safety here:
```
let _ = broker_client.publish_one(msg).await;
```

Everything else should be safe.

Reply:
Then let's just put a reminder comment here.
(This method is just way too long, that's why I'd prefer the reminders)

```suggestion

            Some(()) = async {
                // Reminder: this match arm needs to be cancellation-safe.
```

Reply:
Then let's just put a reminder comment here.
(This method is just way too long, that's why I'd prefer the reminders)

```suggestion

            Some(()) = async {
                // Reminder: this match arm needs to be cancellation-safe.
```